addons/html_editor/static/src/main/table/table_plugin.js

```diff
@@ -561,13 +561,40 @@ export class TablePlugin extends Plugin {
         const selection = this.dependencies.selection.getSelectionData().deepEditableSelection;
         const startTable = closestElement(selection.anchorNode, "table");
         const endTable = closestElement(selection.focusNode, "table");
-        if (!(startTable && endTable) || startTable !== endTable) {
+        if (!(startTable || endTable)) {
             return;
         }
         const [startTd, endTd] = [
             closestElement(selection.anchorNode, "td"),
             closestElement(selection.focusNode, "td"),
         ];
+        if (startTable !== endTable) {
+            // Deselect the table at once if it is fully selected.
```

Comment:
Change to `// Deselect the table if it was fully selected.`, otherwise it reads like it is impossible to fully select a table as it will be automatically deselected due to the use of present tense with "at once".